src/Tracking/DisasterTrackerApp.Dal/Repositories/Implementation/DisasterEventRepository.cs

```diff
@@ -23,4 +23,11 @@ public async Task<List<DisasterEvent>> GetDisasterEventsFiltered(Expression<Func
     {
         return await _context.DisasterEvent.Where(predicate).ToListAsync();
     }
+    public async Task AddExceptClosedDisasterEvents(IEnumerable<DisasterEvent> disasterEvents)
+    {
+        var oldClosedEvents = _context.DisasterEvent.ToList();
+        var newClosedEvents = disasterEvents.Where(x => oldClosedEvents.All(y => y.ExternalApiId != x.ExternalApiId));
+        await _context.DisasterEvent.AddRangeAsync(newClosedEvents);
+        await _context.SaveChangesAsync();
+    }
```

Comment:
not sure if it works properly
but it definitely needs some performance improvements
_context.DisasterEvent.ToList(); - load all data in memory sync....
.Where(x => oldClosedEvents.All(y => y.ExternalApiId != x.ExternalApiId)) - filtering large amoun of data in memory

you can do it with db engine